tsfc/driver.py

```diff
@@ -196,12 +178,13 @@ def facetarea():
 
         # Check if the integral has a quad degree attached, otherwise use
         # the estimated polynomial degree attached by compute_form_data
-        quadrature_degree = params.get("quadrature_degree",
-                                       params["estimated_polynomial_degree"])
+        quad_degree = params.get("quadrature_degree")
+        if quad_degree in [None, "auto", "default", -1, "-1"]:
```

Comment:
I would prefer to have this great variety of equivalent default values reduced to one in the `tsfcrepr` part of FFC, before calling this function.